controller/src/main/java/io/pravega/controller/server/bucket/BucketService.java

```diff
@@ -69,40 +76,33 @@
     private final int bucketId;
     @Getter(AccessLevel.PROTECTED)
     private final BucketStore.ServiceType serviceType;
-    private final Semaphore semaphore;
+    private int avaiableSlots;
```

Comment:
This field is not thread safe. Either make `AtomicInteger` or use `synchronized`(and `GuardedBy`)

Reply:
It is protected by "lock" and made threadsafe. i have just not marked it as such with the GuardedBy annotation. Will update it. thanks